### ISSUE TITLE
Muons: Lakeshore 372 temperature controller

### ISSUE THREAD

**Issue by Tom-Willemsen** [ISISComputingGroup/IBEX#5103]
As a user on EMU using a triton dilution fridge, I need to be able to control the temperature via a Lakeshore 372 temperature controller. This is a standalone box and is not connected to the triton control PC. It is only used on the muon beamlines.

### Acceptance criteria:
- Can get and set the following parameters from "Channel A" on a Lakeshore 372:
  * Temperature
  * PID settings
  * Heater range
- Can get the following parameters from "Channel A" on a Lakeshore 372:
  * Heater power
  * Sensor resistance

### Notes:
- A triton will be being run on EMU using this setup on the 24th Feb
- Only readings from "Channel A" are required - there are a number of other channels which the lakeshore can be set up to scan, but they are not used.
- Temperature calibration curves are put into the lakeshore manually - this does not need to be done in software.
- I have added a manual at `\\isis\shares\ISIS_Experimental_Controls\Manuals\Lakeshore__temperature_controller\Model_372`
- There is a labview driver available at `C:\LabVIEW Modules\Drivers\Lakeshore 372`
- From a brief look at the manual and labview VI, it looks like a relatively straightforwards stream device IOC - similar to our other lakeshore temperature controller IOCs.
- This setup is specific to the muons, and was originally tested on HIFI (our work with this setup is detailed in https://github.com/ISISComputingGroup/IBEX/issues/5010 )

**Comment by kjwoodsISIS** [comment]
Clarification from Steve Cottrell:
> Hi Kevin,
>
> Sorry not to have picked up on that. The muon Triton fridges have an additional Lakeshore 
> controller (the 372) to allow control and continuous readout of the sample temperature 
> (the OI software doesn’t allow this using the in-built Lakeshore, despite the unit having 
> a dedicated channel for the sample thermometry).
>
> The Triton fridge will be run starting Monday 24th Feb. However, this is commissioning time, 
> so we can easily accommodate some testing during this period. There is a user starting with 
> the fridge on the 28th Feb, immediately following the commissioning time.
>
>Steve

**Comment by Tom-Willemsen** [comment]
PRs:
- https://github.com/ISISComputingGroup/EPICS/pull/199
- https://github.com/ISISComputingGroup/EPICS-lakeshore372/pull/1
- https://github.com/ISISComputingGroup/EPICS-ioc/pull/471
- https://github.com/ISISComputingGroup/EPICS-IOC_Test_Framework/pull/291
- https://github.com/ISISComputingGroup/EPICS-DeviceEmulator/pull/65
- https://github.com/ISISComputingGroup/ibex_gui/pull/1041

**Comment by DominicOram** [comment]
Can you update https://epics-controls.org/resources-and-support/modules/hardware-support/#participants-list-1 if not already done?

**Comment by Tom-Willemsen** [comment]
Our wiki states that we should only do that once the ticket is merged - but yes once merged I will add it.
